pageserver/src/pgdatadir_mapping.rs

```diff
@@ -1914,25 +1952,26 @@ impl<'a> DatadirModification<'a> {
 
     fn put(&mut self, key: Key, val: Value) {
         if key.is_rel_block_key() || key.is_slru_block_key() {
-            self.put_data(key, val)
+            self.put_data(key.to_compact(), val)
         } else {
-            self.put_metadata(key, val)
+            self.put_metadata(key.to_compact(), val)
         }
     }
 
-    fn put_data(&mut self, key: Key, val: Value) {
-        // TODO: old code had logic for eliding multiple updates at the same lsn to the same key: was that really
-        // needed for data pages?
```

Comment:
So the elisions were for zero pages?

Reply:
Right.  I checked with the compute team that they don't expect postgres itself to ever write the same (key, lsn) twice: these are just originating from the synthetic zero writes the pageserver generates.